internal/riverinternaltest/testfactory/test_factory.go

```diff
@@ -111,3 +117,26 @@ var seq int64 = 1 //nolint:gochecknoglobals
 func nextSeq() int {
 	return int(atomic.AddInt64(&seq, 1))
 }
+
+type QueueOpts struct {
+	Metadata  []byte
+	Name      *string
+	UpdatedAt *time.Time
+}
+
+func Queue(ctx context.Context, tb testing.TB, exec riverdriver.Executor, opts *QueueOpts) *rivertype.Queue {
+	tb.Helper()
+
+	metadata := opts.Metadata
+	if opts.Metadata == nil {
+		metadata = []byte("{}")
+	}
+
+	queue, err := exec.QueueCreateOrSetUpdatedAt(ctx, &riverdriver.QueueCreateOrSetUpdatedAtParams{
+		Metadata:  metadata,
+		Name:      ptrutil.ValOrDefault(opts.Name, "default"),
```

Comment:
Could you change this to?

``` go
		Name:      ptrutil.ValOrDefaultFunc(opts.Name, func() string { return fmt.Sprintf("queue_%02d", nextSeq()) }),
```

The basic idea with these factory functions is you can call them many times with no (or very few) opts and still get a new valid object back. See migrations above for example, which _could_ always returning migration version 1, but generates a sequence instead for caller convenience.

This would also allow you to simplify your tests quite a bit, many of which have to override `Name` to work around this default.